index.d.ts

```diff
@@ -0,0 +1,398 @@
+declare module 'pc-nrfconnect-shared' {
+    import { Reducer, AnyAction } from 'redux';
+    import React from 'react';
+    import winston from 'winston';
+
+    // State
+
+    interface NrfConnectState<AppState> {
```

Comment:
Is the idea to keep these types in a separate file and not convert the normal JavaScript files to TypeScript, so that the types are defined (and used!) were they originate?

Reply:
I think long-term we should move the original files to TypeScript, in which case this file would gradually become unnecessary. For now though, this was a nice way to put all the types in one place, without having to touch every file in the project. It's the approach used by most JavaScript libraries that need to be interfaced with from TypeScript.